src/Sarif.Viewer.VisualStudio.Core/SdkUiUtilities.cs

```diff
@@ -141,6 +142,18 @@ internal static IVsWindowFrame OpenDocument(IServiceProvider provider, string fi
                 return null;
             }
 
+            // prompt user if open binary file in editor
+            if (promptForBinFile)
+            {
+                using (FileStream stream = File.OpenRead(file))
+                {
+                    if (IsBinaryFile(stream) && !AllowOpenBinaryFile(file))
+                    {
+                        return null;
+                    }
+                }
```

Comment:
you have a flag that you check if you want to prompt or not. How do u define this?

Also, here is what u are doing:
1. opening/reading the file
2. checking if that is a binary file
3. checking the extension and prompting the user.

shouldn't you do the checking + prompting first? If the user says `no`, so you dont need to open/check the binary, for example.

Reply:
The logic is in method AllowOpenBinaryFile. In latest change I updated the flow now its:
1. Checking if file extension is already allowed.
    a. If allowed, return true.
    b. If not allowed, go to step 2.
2. Checking if file is binary file.
    a. If yes then go to step 3.
    b. If no return true.
3. Prompting user if open the binary in VS.

Here the reason why checking if the file is binary file before prompting user, is that we cannot determine whether a file is binary or not according to file extension. E.g. file with extension txt but content is binary, or a unknow type of file extension. Please let me know what do you think